### PR TITLE
feat(scan): mobile matching view — official section + equivalents + low_confidence (Issue #700)

### LINKED ISSUE #700: [Scan T2 — #5] Mobile UI: matching view (official section + top 3)
## Context

Chunk #5 of Scan Tranche 2. Renders the output of the matching algorithm (chunk #4) as the two distinct sections defined in brainstorm §2: **"🏆 Brewery recipe"** + **"🧪 Equivalent recipes"** (top 3).

The pharmacy metaphor from the brainstorm drives the visual distinction: the official recipe is the brand-name medicine (prominent, badged), the equivalents are generics (uniform, ranked).

### Sections

- [ ] **"🏆 Brewery recipe"** — single card at the top, visually distinct (brand badge + gold accent). Hidden if no official recipe exists for this beer.
- [ ] **"🧪 Equivalent recipes"** — horizontal list / stacked cards of the top 3 equivalent recipes returned by chunk #4.
- [ ] **"See more"** button below the top 3 → navigates to a full list screen (can be a simple filtered list for v0.1, no elaborate UI).

### Low-confidence warning

- [ ] When the API response carries `low_confidence: true`, display a discreet warning *"Aucune recette très similaire dans la base. Voici les plus proches."* above the equivalent recipes section.

### Recipe card content

- [ ] Recipe name + author.
- [ ] Rating (stars) + brew count.
- [ ] Style.
- [ ] Tap → navigates to the recipe detail (existing screen).

### Import flow

- [ ] A CTA "Importer dans Mes Recettes" on any recipe card creates a local copy in the user's recipes list. Implementation can stub the persistence for v0.1 as long as the UX feels real.

### Tests

- [ ] Unit: official section hidden when no official recipe.
- [ ] Unit: low-confidence warning shown only when flag is present.
- [ ] Unit: "See more" navigates correctly.
- [ ] Integration: import flow produces a new entry in My Recipes.

## Out of scope

- Full list screen UI beyond a basic filtered list (v0.1).
- Community rating submission (v0.2+).

## Effort estimate

**1 focused dev day**.

## Dependencies

- **Blocked by** [#698 — Scan T2 #3 info card UI](../issues/698) (for layout consistency).
- **Blocked by** [#699 — Scan T2 #4 matching algo](../issues/699) (provides the data).

## References

- [Scan brainstorm 2026-04-24 §2 UX + §3 matching](../blob/main/docs/product/brainstorms/scan-2026-04-24.md)
- Pharmacy metaphor — §1 of brainstorm.

### PR DESCRIPTION
**Pull request by benoit-bremaud**
## Summary

- Add `ScanMatchingResult` envelope type (`rankings` + `lowConfidence`) and `isOfficial`/`style` fields to `ScanRecipeMatch` in domain
- New `recipe-matching.api.ts` data layer: fetches `GET /recipes/match/:beerId`, maps `MatchingResponseWireDto` (snake_case) → `ScanMatchingResult` (camelCase)
- New `recipe-matching.use-cases.ts`: branches on `dataSource.useDemoData` — demo path wraps `getDemoEquivalentRecipes()` in the envelope; backend path delegates to `fetchMatchingRecipes()`
- `BeerInfoCardScreen`: full refactor of the equivalents section into `MatchingRecipesSection` — splits official recipe (🏆 gold accent) above community alternatives (🧪 capped at 3); renders a low-confidence notice when `matching.lowConfidence === true`; loading state + error handling
- Add `warningBackground` design token to `colors.ts`
- 4 new use-case tests (demo happy/sad + backend happy/sad); 7 new presentation tests (envelope scenarios: low_confidence, official section presence/absence, cap at 3, empty state, API error)
- `website/docs/ROADMAP.md` Phase 5 updated with #700 entry

## Checklist

- [x] Happy path + sad path + edge cases covered
- [x] `tsc --noEmit` passes
- [x] Build passes
- [x] Existing tests still green (560 / 560)
- [x] No `any`, no inline styles introduced

Closes #700

### CODE REVIEW

**Comment by benoit-bremaud** [comment]
@Smith06S @Thais9723 — For your information: PR #793 adds the recipe matching view to the scan screen.

When a beer is scanned, the app now shows two distinct sections: one for the official reference recipe (if any exists in our catalog) and one for the top community alternatives (limited to 3). A low-confidence notice appears when none of the matches are a strong fit.

This connects to the matching algorithm already in production (Issue #699). The demo data path is fully functional for testing without a backend connection.

560 tests passing, CI green.

File: packages/mobile-app/src/features/scan/presentation/BeerInfoCardScreen.tsx [hunk 1]
```diff
@@ -295,55 +338,127 @@ function EquivalentRecipesSection({
...
+            isImporting={importingId === officialRecipe.recipeId}
+            isDisabled={importingId !== null}
+            onPress={() => handleImport(getImportSourceId(officialRecipe))}
+            highlightOfficial
+          />
+        </Card>
+      ) : null}
+
+      {equivalentRecipes.length > 0 ? (
+        <Card style={styles.recipesCard}>
```
[the remark this answers is not in the document]
Comment (benoit-bremaud) [reply packages/mobile-app/src/features/scan/presentation/BeerInfoCardScreen.tsx:391]: **P2 — Implemented** (commit d8f4bd8)

The low-confidence warning has been moved out of the `equivalentRecipes.length > 0` conditional block and is now rendered directly inside the top-level `<View>`, between the official card and the equivalents card. It surfaces whenever `matching.lowConfidence === true`, regardless of whether any non-official equivalents are present.

A new test covers the edge case you identified: API returns a single `isOfficial=true` recipe with `lowConfidence: true` → warning still appears even though `equivalentRecipes` is empty.

**Comment by benoit-bremaud** [comment]
**Review summary — PR #793 ready to merge**

CI: all checks green (mobile-app, SonarCloud, security-audit, website — 3 runs).

Codex review: 1 comment (P2) — addressed in commit d8f4bd8. No new comments after re-review.

Fixes pushed on top of the feature commit:
- 0cdcaab — async timing flake in BeerInfoCardScreen test (getByText → findByText for matching section)
- d8f4bd8 — low_confidence warning moved outside equivalents block + new edge-case test (Codex P2)
- 6eeddf9 — Prettier format pass

22 BeerInfoCardScreen tests passing (was 14 before #700). 560 total across the mobile-app suite.
